src/Utility/RerunRuntimeSuiteTester.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace SilverStripe\BehatExtension\Utility;
+
+use Behat\Testwork\Environment\Environment;
+use Behat\Testwork\Specification\SpecificationIterator;
+use Behat\Testwork\Tester\Result\IntegerTestResult;
+use Behat\Testwork\Tester\Result\TestResult;
+use Behat\Testwork\Tester\Result\TestResults;
+use Behat\Testwork\Tester\Result\TestWithSetupResult;
+use Behat\Testwork\Tester\Setup\SuccessfulSetup;
+use Behat\Testwork\Tester\Setup\SuccessfulTeardown;
+use Behat\Testwork\Tester\SpecificationTester;
+use Behat\Testwork\Tester\SuiteTester;
+
+/**
+ * Copy paste of Behat\Testwork\Tester\Runtime\RuntimeSuiteTester which is a final class
+ *
+ * Reruns failed features
+ */
+class RerunRuntimeSuiteTester implements SuiteTester
```

Comment:
In the issue you [mentioned](https://github.com/silverstripe/silverstripe-admin/issues/1671#issuecomment-2041832665) that you were able to retry scenarios from within the `IsolatingScenarioTester`.

Just confirming here that there wasn't a clean way to fully reset from within that scope, which is why we're doing it at this level?

Reply:
Redoing at that level wasn't redoing the background step, meant you were still logged in as an admin if you logged in during the first failed run, etc